gettext-maps/build.rs

```diff
@@ -0,0 +1,149 @@
+
+    for dir_entry_result in po_dir.read_dir().unwrap() {
+        let dir_entry = dir_entry_result.unwrap();
+        let po_file_path = dir_entry.path();
+        let lang = po_file_path
+            .file_stem()
+            .expect("All entries in the po directory must be regular files.");
+        let language = lang.to_str().unwrap().to_owned();
+
+        if !all_langs_enabled
+            && std::env::var(format!("CARGO_FEATURE_{}", language.to_uppercase())).is_err()
+        {
+            // Localizations for this language are not enabled.
+            // The `CARGO_FEATURE_*` variables are set by cargo based on which features are enabled.
+            // The feature names are converted to uppercase and `-` is replaced by `_`.
+            continue;
+        }
+        let mo_file_path = mo_file_dir.join(lang);
+
+        let update_mo_file = || {
+            // Try to create a new MO file.
+            std::process::Command::new("msgfmt")
+                .arg("--check-format")
+                .arg(format!("--output-file={}", mo_file_path.to_str().unwrap()))
+                .arg(&po_file_path)
+                .output()
+        };
+
+        if let Ok(metadata) = std::fs::metadata(&mo_file_path) {
+            // MO file exists, but might be outdated.
+            let mo_mtime = metadata.modified().unwrap();
+            let po_mtime = dir_entry.metadata().unwrap().modified().unwrap();
+            if mo_mtime < po_mtime {
+                update_mo_file().unwrap();
+            };
+        } else {
+            // MO file does not exist.
+            update_mo_file().unwrap();
+        }
+
+        // At this point we have an up-to-date MO file.
+        // We proceed to parse the file into a map, which is put into a Rust source file,
+        // such that it can be used from other source code.
+
+        let mut mo_file = File::open(&mo_file_path).unwrap();
+        let mut file_content = vec![];
+        mo_file.read_to_end(&mut file_content).unwrap();
+        let language_localizations = parse_mo_file(&file_content).unwrap();
+
+        let mut single_language_localization_map = phf_codegen::Map::new();
+
+        // The values will be written into the source code as is, meaning escape sequences and
+        // double quotes in the data will be interpreted by the Rust compiler, which is undesirable.
+        // Converting them to raw strings prevents this. (As long as no input data contains `"###`.)
+        fn to_raw_str(s: &str) -> String {
+            format!("r###\"{s}\"###")
+        }
+        for (k, v) in language_localizations {
+            single_language_localization_map.entry(
+                String::from_utf8(k.into()).unwrap(),
+                to_raw_str(&String::from_utf8(v.into()).unwrap()),
+            );
+        }
+        // TODO(MSRV >= 1.74): remove this. Workspace lints allow non_upper_case_globals, but
+        // workspace lints are not supported prior to Rust 1.74.
```

Comment:
not sure about "remove this" - I think we want to get rid of the
workspace-level `allow(non_upper_case_globals)`, then this would stay